doc/source/contributing.rst

```diff
@@ -221,14 +218,12 @@ You'll need to have at least python3.5 installed on your system.
    . ~/virtualenvs/pandas-dev/bin/activate
 
    # Install the build dependencies
-   python -m pip install -r ci/requirements_dev.txt
```

Comment:
to be honest I would rather just ``conda env create -f environment.yaml`` (at the very beginning) than do a pip install here. This is a bit archaic.

Reply:
As said, happy to remove `requirements.txt` and the set up instructions for pip if there is agreement on that. It's my preferred option too.

I added them motivated by @TomAugspurger last paragraph in this comment: https://github.com/pandas-dev/pandas/issues/23115#issuecomment-435945502

Reply:
I am also in favor of keeping the pip option as well. 

One thing, I would maybe call it `dev-requirements.txt` instead of `requirements.txt` (or the original `requiements_dev.txt`), to eliminate possible confusion with all other use cases of a "requirements" file.

And maybe the same for `dev-environment.yml`, that groups the two also nicely together.